src/video_utils.py

```diff
@@ -206,45 +271,61 @@ def download_video(url):
                 break  # Exit the loop once the file is found
     except subprocess.CalledProcessError as e:
         error("Error downloading video: %s", e)
+        if "[Instagram]" in str(e) and "No video formats found!" in str(e):
+            debug("The yt-dlp did not find Instagram reels, trying gallery-dl for images")
+        try:
+            result_path = download_instagram_media(url, temp_dir)
+            if result_path:
+                debug("Successfully downloaded Instagram media using gallery-dl")
+                return result_path
+            else:
+                error("Failed to download Instagram media using gallery-dl")
+        except Exception as gallery_dl_error:  # pylint: disable=broad-except
+            error("Unexpected error during Instagram download images by gallery-dl: %s", gallery_dl_error)
     except subprocess.TimeoutExpired as e:
         error("Download process timed out: %s", e)
     except (OSError, IOError) as e:
-        debug("Downloading video from URL: %s", url)
         error("File system error occurred: %s", e)
     except yt_dlp.utils.DownloadError as e:
         error("Download error occurred: %s", e)
     except yt_dlp.utils.ExtractorError as e:
         error("Extractor error occurred: %s", e)
+    finally:
+        if result_path is None:
+            shutil.rmtree(temp_dir)
 
     return result_path  # Return the result variable at the end
 
 
-def cleanup(video_path):
+def cleanup(media_path):
     """
     Cleans up temporary files by deleting the specified video file and its containing directory.
 
-    This function attempts to remove the specified video file and
+    This function removes the specified media_path with video or images and
     its parent directory. Logs are printed if debugging is enabled.
 
     Parameters:
-        video_path (str): The path to the video file to delete.
+        media_path (list): The path to the video file to delete.
 
     Logs:
         Logs messages about the deletion process or any errors encountered.
     """
-    debug("Temporary directory to delete %s", video_path)
 
-    if os.path.isdir(video_path):
-        debug("Temporary directory is empty: %s", video_path)
-        temp_dir = video_path
-    else:
-        debug("Temporary directory is not empty: %s", os.path.dirname(video_path))
-        temp_dir = os.path.dirname(video_path)
+    folder_to_delete = None
+    if isinstance(media_path, list):
```

Comment:
if `media_path` is not a list then `shutil.rmtree` will try to delete `None`? Which will cause `Except`. 
This function could be adjusted to handle `temp_dir` removal other than remove it in `download_media` function

Reply:
With the current logic, it will not delete anything, however, when files are downloaded (doesn't matter with the `gallery-dl` or `yt-dlp`), media_path will always be a list